src/psyclone/domain/gocean/nodes/gocean_extract_node.py

```diff
@@ -141,66 +142,111 @@ def create_driver(self, input_list, output_list):
+                       .format(psy_data, module_name, region_name))
         prog.add(call)
 
         post_suffix = self._post_name
+
+        # Variables might need to be renamed in order to guarantee unique
+        # variable names in the driver: An example of this would be if the
+        # user code contains a variable 'dx', and the kernel takes a
+        # property 'dx' as well. In the original code that is no problem,
+        # since the property is used via field%grid%dx. But the stand-alone
+        # driver renames field%grid%dx to dx, which can cause a name clash.
+        # Similar problems can exist with any user defined type, since all
+        # user defined types are rewritten to just use the field name.
+        # We use a mapping to support renaming of variables: it takes as
+        # key the variable as used in the original program (e.g. 'dx' from
+        # an expression like field%grid%dx), and maps it to a unique local
+        # name (e.g. dx_0).
+
+        rename_variable = {}
         for var_name in all_vars:
             # TODO #644: we need to identify arrays!!
+            # Support GOcean properties, which are accessed via a
+            # derived type (e.g. 'fld%grid%dx'). In this stand-alone
+            # driver we don't have the derived type, instead we create
+            # variable based on the field in the derived type ('dx'
+            # in the example above), and pass this variable to the
+            # instrumented code.
+            last_percent = var_name.rfind("%")
+            if last_percent > -1:
+                # Strip off the derived type, and only leave the last
+                # field, which is used as the local variable name.
+                local_name = var_name[last_percent+1:]
+            else:
+                # No derived type, so we can just use the
+                # variable name directly in the driver
+                local_name = var_name
+            unique_local_name = sym_table.new_symbol_name(local_name)
+            rename_variable[local_name] = unique_local_name
+            sym_table.add(Symbol(unique_local_name))
+            local_name = unique_local_name
+
+            # TODO: #644 - we need to identify arrays!!
             # Any variable used needs to be defined. We also need
             # to handle the kind property better and not rely on
             # a hard-coded value.
-            decl = DeclGen(prog, "real", [var_name], kind="8",
+            decl = DeclGen(prog, "real", [local_name], kind="8",
                            dimension=":,:", allocatable=True)
             prog.add(decl)
             is_input = var_name in input_list
             is_output = var_name in output_list
 
             if is_input and not is_output:
                 # We only need the pre-variable, and we can read
-                # it from the file (and allocate its size)
+                # it from the file (which also allocate space for it)
```

Comment:
Suggest: "... (this call also allocates space for it)." without the quotes.

Reply:
Done.